untrusted/test_enclave/build.rs

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+#![doc = include_str!("README.md")]
+
+extern crate bindgen;
+use cargo_emit::rerun_if_changed;
+use std::env;
+use std::path::{Path, PathBuf};
+use cc::Build;
+use std::process::{Command};
+
+/// Edger generated C files.  This is only the C files.
+struct EdgerFiles {
+    /// The full path to the trusted C file, <path/to/basename>_t.c
+    trusted: PathBuf,
+
+    /// The full path to the untrusted C file, <path/to/basename>_u.c
+    untrusted: PathBuf
+}
+
+const DEFAULT_SGX_SDK_PATH: &str = "/opt/intel/sgxsdk";
+const EDGER_FILE: &str = "src/enclave.edl";
+const ENCLAVE_FILE: &str = "src/enclave.c";
+const ENCLAVE_LINKER_SCRIPT: &str = "src/enclave.lds";
+const SIGNING_KEY: &str = "src/signing_key.pem";
+const ENCLAVE_CONFIG: &str = "src/config.xml";
+
+fn main() {
+    let root_dir = root_dir();
+    let edger_files = build_enclave_definitions(root_dir.join(EDGER_FILE));
+
+    build_enclave_binary([root_dir.join(ENCLAVE_FILE), edger_files.trusted]);
+    build_untrusted_library(&edger_files.untrusted);
+
+    let mut untrusted_header = edger_files.untrusted.clone();
+    untrusted_header.set_extension("h");
+    build_untrusted_bindings(untrusted_header);
+}
+
+/// Provide the base path for the Intel SGX SDK.  Will use the environment
+/// variable `SGX_SDK`.  If this isn't set it will default to
+/// `/opt/intel/sgxsdk`.
+fn sgx_library_path() -> String {
+    env::var("SGX_SDK").unwrap_or_else(|_| DEFAULT_SGX_SDK_PATH.into())
+}
+
+/// The value of the environment variable `OUT_DIR`, this must be set.
+/// See https://doc.rust-lang.org/cargo/reference/environment-variables.html
+fn out_dir() -> PathBuf {
+    env::var("OUT_DIR").unwrap().into()
+}
+
+/// The root dir of this crate. Will be the value of `CARGO_MANIFEST_DIR`
+/// See https://doc.rust-lang.org/cargo/reference/environment-variables.html
+fn root_dir() -> PathBuf {
+    env::var("CARGO_MANIFEST_DIR").unwrap().into()
+}
+
+/// The ld linker to use.  This has to be an ld linker as lld will fail
+/// to link a working enclave
+fn ld_linker() -> String {
+    env::var("LD").unwrap_or_else(|_| "ld".into())
+}
+
+fn sgx_library_suffix() -> &'static str {
+    match () {
+        #[cfg(feature = "hw")]
+        () => "",
+        #[cfg(not(feature = "hw"))]
+        () => "_sim",
+    }
```

Comment:
The `match` seems unnecessary here
```suggestion
    #[cfg(feature = "hw")]
    ""
    #[cfg(not(feature = "hw"))]
    "_sim"
```

Or better yet, make it a constant (with 2 `cfg`-gated declarations)

Reply:
I haven't had a chance to look up the syntax issue
```diff
index 9f964ad..554a37b 100644
--- a/untrusted/urts_sys/build.rs
+++ b/untrusted/urts_sys/build.rs
@@ -12,12 +12,10 @@ fn sgx_library_path() -> String {
 }
 
 fn sgx_library_suffix() -> &'static str {
-    match () {
         #[cfg(feature = "hw")]
-        () => "",
+        ""
         #[cfg(not(feature = "hw"))]
-        () => "_sim",
-    }
+        "_sim"
 }
 
```

```
➜  untrusted git:(feature/untrusted) ✗ cargo test
   Compiling mc-sgx-urts-sys v0.1.0 (/home/nick/git/sgx/untrusted/urts_sys)
   Compiling test_enclave v0.1.0 (/home/nick/git/sgx/untrusted/test_enclave)
error: expected one of `.`, `;`, `?`, `}`, or an operator, found `#`
  --> urts_sys/build.rs:17:9
   |
16 |         ""
   |           - expected one of `.`, `;`, `?`, `}`, or an operator
17 |         #[cfg(not(feature = "hw"))]
   |         ^ unexpected token

error: could not compile `mc-sgx-urts-sys` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```

Converted to constants